GitLab Command Set Init (#59)
nimbella/command-sets: gitlab/README.md

# Nimbella Commander GitLab Command Set
A Nimbella Commander command set for viewing GitLab project statistics, creating tickets, and assigning issues.

## Features
- View community contribution stats for GitLab projects
- Create a ticket for a GitLab project
- Assign issues for a GitLab project
- View list of users in project repository

## Install
```

> /nc csm_install github:Maljean/command-sets/tree/dev/gitlab

Reviewer: This needs to be just `/nc csm_install gitlab`. The dev command-sets are deployed to dev namespace where `csm_install gitlab` works. The above command what you typed works when as well but we have to assume we are going to be running in production.

Author: That's neat, thank you. I'll keep that in mind for future command sets.

```

## Requirements
```

> In order to use this command set, you must set up a Personal Access Token on GitLab with API access. And save the token as a secret with the key 'AcessToken_GitLab'

Reviewer 2: ```suggestion
In order to use this command set, you must set up a Personal Access Token on GitLab with API access. And save the token as a secret with the key 'gitlabToken'
```

Author: Can do 👍

Use /nc secret_create to create keys
Visit https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html to learn how to setup a Personal Access Token on GitLab
```

## Usage
Viewing community contributions for a GitLab project.
```
/dapp gitlab_stats <repo>
```

> ![](https://raw.githubusercontent.com/Maljean/command-sets/master/gitlab/screenshots/statsCommandScreenshot.PNG)

Reviewer: Should the github url be updated to point to `command-set` repo instead?

Author: It should! Last week I was very hesitant to guess it. Now that I know this will be on the master branch I can update it.

- Assign issues for a GitLab project
- View list of users in project repository

## Install
```
/nc csm_install gitlab
```

## Requirements
```
In order to use this command set, you must set up a Personal Access Token on GitLab with API access. And save the token as a secret with the key 'gitlabToken'
Use /nc secret_create to create keys
Visit https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html to learn how to setup a Personal Access Token on GitLab
```

## Usage
Viewing community contributions for a GitLab project.
```
/dapp gitlab_stats <repo>
```
![](https://raw.githubusercontent.com/nimbella/command-sets/tree/master/gitlab/screenshots/statsCommandScreenshot.PNG)

Creating a ticket for a GitLab project.
```
/dapp gitlab_create <repo> <title> <description>
```
![](https://raw.githubusercontent.com/nimbella/command-sets/tree/master/gitlab/screenshots/createCommandScreenshot.PNG)

Viewing list of users in project repository.
```
/dapp gitlab_users <repo> [<name>]
```
![](https://raw.githubusercontent.com/nimbella/command-sets/tree/master/gitlab/screenshots/usersCommandScreenshot.PNG)

Viewing GitLab command set documentation.
```

> /dapp gitlab_gitlab

Reviewer: Should just be `/dapp gitlab`

# Nimbella Commander GitLab Command Set
A Nimbella Commander command set for viewing GitLab project statistics, creating tickets, and assigning issues.

## Features
- View community contribution stats for GitLab projects
- Create a ticket for a GitLab project
- Assign issues for a GitLab project
- View list of users in project repository

## Install
```
/nc csm_install gitlab
```

## Requirements
```
In order to use this command set, you must set up a Personal Access Token on GitLab with API access. And save the token as a secret with the key 'gitlabToken'
Use /nc secret_create to create keys
Visit https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html to learn how to setup a Personal Access Token on GitLab
```

## Usage
Viewing community contributions for a GitLab project.
```
/dapp gitlab_stats <repo>
```

> ![](https://raw.githubusercontent.com/nimbella/command-sets/tree/master/gitlab/screenshots/statsCommandScreenshot.PNG)

Reviewer 2: ```suggestion
![GitLab stats command](screenshots/statsCommandScreenshot.PNG)
```
The links can be relative.

Creating a ticket for a GitLab project.
```
/dapp gitlab_create <repo> <title> <description>
```

> ![](https://raw.githubusercontent.com/nimbella/command-sets/tree/master/gitlab/screenshots/createCommandScreenshot.PNG)

Reviewer 2: ```suggestion
![GitLab create command](screenshots/createCommandScreenshot.PNG)
```

Viewing list of users in project repository.
```
/dapp gitlab_users <repo> [<name>]
```

> ![](https://raw.githubusercontent.com/nimbella/command-sets/tree/master/gitlab/screenshots/usersCommandScreenshot.PNG)

Reviewer 2: ```suggestion
![GitLab users command](screenshots/usersCommandScreenshot.PNG)
```

Viewing GitLab project issues.
Note: Status may be all, opened, or closed.
```
/dapp gitlab_issues

> ```

Reviewer 2: It would be nice to have screenshots for other commands as well.

Author: Indeed. I'll upload some soon
